Convert API client to use AWS Go SDK

euank opened the pull request:
This replaces the generated client with the official AWS Go SDK's client.
This also improves testing around the api client as a bonus.

This will make it easier to update the client in the future, provides better connection pooling (via using http.Client directly), and hopefully does not have any major regressions.

Currently known regression: the '-k' flag does not work correctly with this. I'll add a commit that fixes that to this request before I merge it, but feel free to review this now.

aaithal commented on agent/api/api_client.go:
```diff
@@ -14,158 +14,112 @@
+func getCpuAndMemory() (int64, int64) {
 	memInfo, err := system.ReadMemInfo()
-	mem := int32(memInfo.MemTotal / 1024 / 1024) // MB
+	mem := memInfo.MemTotal / 1024 / 1024 // MB
```
Nit: This is MiB, not MB

samuelkarp commented on agent/api/api_client.go:
```diff
@@ -14,158 +14,112 @@
 
+// ECSClient is an interface over the ECSSDK interface which abstracts away some
+// details around constructing the request and reading the response down to the
+// parts ECS cares about.
```
s/ECS/the agent/

samuelkarp commented:
:+1: 
This looks really good other than the two trivial comments above (mine and @aaithal's).

euank commented:
@samuelkarp @aaithal I addressed your comments, corrected the retry logic, and plumbed through the -k flag.

The retry logic should be revisited again because the sdk should handle that correctly soon, but even if there's two layers of retries that's not all that big a deal.
